### ISSUE TITLE
Bender clears selection

### ISSUE THREAD

**Issue by Comandeer** [ckeditor/ckeditor-dev#839]
## Are you reporting a feature request or a bug?

Task (bug in test tools)

## Provide detailed reproduction steps (if any)

1. Open https://jsfiddle.net/Comandeer/nwkv3bn5/

### Expected result

`Test` should be selected.

### Actual result

Selection is collapsed at the beginning of editor's content.

## Other details

* Browser: Chrome 60.0.3112.113

It seems that creating bookmarks inside `bender.tools.getHtmlWithSelection` causes the issue, however only if `bender.tools.setHtmlWithSelection` was used earlier. Setting selection via manually created ranges is working fine.

Note also that `getHtmlWithSelection` is returning correct result.

**Comment by msamsel** [comment]
**TL;DR** -> Use `bender.tools.selection.getWithHtml` instead of `bender.tools.getHtmlWithSelection`

----
I've made a research with this issue and it seems to be much more complicated that it looks. Below few information which I find out.
1. `bender.tools.getHtmlWithSelection` and `bender.tools.setHtmlWithSelection` are [deprecated](https://github.com/ckeditor/ckeditor-dev/blob/master/tests/_benderjs/ckeditor/static/tools.js#L535-L541). It's recommended to not use it any longer.
1. `bender.tools.getHtmlWithSelection` operate on real DOM structure modifying it during entire process.

That's why better choice is to use `bender.tools.selection.getWithHtml`, which operate on clones of elements.

----
Some fixes which I test:
1. Modification of `bender.tools.getHtmlWithSelection` to use `bender.tools.selection.getWithHtml` inside it.
Result: this kind of code modification return few hundreds failed test cases.
1. Deeper research narrow problem to this part:
  * https://github.com/ckeditor/ckeditor-dev/blob/da389f9f08a8df4562ce14ae02ad215dd510c454/tests/_benderjs/ckeditor/static/tools.js#L577-L579
  * https://github.com/ckeditor/ckeditor-dev/blob/da389f9f08a8df4562ce14ae02ad215dd510c454/core/dom/range.js#L748
  * https://github.com/ckeditor/ckeditor-dev/blob/da389f9f08a8df4562ce14ae02ad215dd510c454/core/dom/range.js#L1854
  * https://github.com/ckeditor/ckeditor-dev/blob/da389f9f08a8df4562ce14ae02ad215dd510c454/core/dom/node.js#L269-L272
After `insertBefore` inside range from example, selection changes in Chrome and it's not available any longer.
I've tried to restore selection by simple add `range.select()` after creating bookmark. This effects failing 2 test suites in Chrome:
    * `tests/plugins/magicline/magicline`
    * `tests/plugins/indent/indent`
Further steps would require modification of tests, what seems to be beyond scope of this case.

Because current situation does not affect actual test and seems to be relevant for new test cases only, I'd like to recommend using `bender.tools.selection.getWithHtml`.
